File: examples/perl/scripts/install-data.pl
#!/usr/bin/perl

use strict;
use warnings;

use VDI;

#
# Parameter Handling
#

my ($vdiID, $inputDir) = @ARGV;

if (not defined $vdiID) {
  VDI::logMessage("VDI ID parameter was not provided!");
  exit VDI::EXIT_CODE_UNEXPECTED_ERROR;
}
if (not defined $inputDir) {
  VDI::logMessage("input directory was not provided!");
  exit VDI::EXIT_CODE_UNEXPECTED_ERROR;
}

VDI::verifyVDIID($vdiID);
VDI::verifyDir($inputDir);

my $db_host = VDI::requireEnv("DB_HOST");
my $db_port = VDI::requireEnv("DB_PORT");
my $db_name = VDI::requireEnv("DB_NAME");
my $db_user = VDI::requireEnv("DB_USER");
my $db_pass = VDI::requireEnv("DB_PASS");

#
# Script Body
#

# TODO: Do the installation!

File: examples/python/scripts/install-data.py
#!/usr/bin/env python3

import sys
import vdi

#
# Input Handling
#

if len(sys.argv) < 3:
    vdi.log_message("called with too few arguments, callers must provide a VDI dataset ID and an input directory")
    sys.exit(vdi.EXIT_CODE_UNEXPECTED_ERROR)

VDI_ID = sys.argv[1]
INPUT_DIR = sys.argv[2]

vdi.verify_dir(INPUT_DIR)

DB_HOST = vdi.require_env("DB_HOST")
DB_PORT = vdi.require_env("DB_PORT")
DB_NAME = vdi.require_env("DB_NAME")
DB_USER = vdi.require_env("DB_USER")
DB_PASS = vdi.require_env("DB_PASS")

#
# Script Body
#

# TODO: call some commands and do the things needed to install a dataset
